applications/FluidDynamicsApplication/custom_utilities/mass_conservation_utility.cpp

```diff
@@ -246,75 +317,97 @@ void MassConservationCheckProcess::ComputeVolumesAndInterface( double& positiveV
             }
         }
     }
-    // assigning the values to the arguments of type reference
-    positiveVolume = pos_vol;
-    negativeVolume = neg_vol;
-    interfaceArea = int_area;
+    mrModelPart.GetCommunicator().SynchronizeCurrentDataToMin(DISTANCE);
+    rPositiveVolume = mrModelPart.GetCommunicator().GetDataCommunicator().SumAll(pos_vol);
+    rNegativeVolume = mrModelPart.GetCommunicator().GetDataCommunicator().SumAll(neg_vol);
+    rInterfaceArea = mrModelPart.GetCommunicator().GetDataCommunicator().SumAll(int_area);
 }
 
 
-double MassConservationCheckProcess::ComputeInterfaceArea(){
 
-    double int_area = 0.0;
+double MassConservationUtility::OrthogonalFlowIntoAir( const double Factor )
+{
+    double outflow = 0.0;
 
-    #pragma omp parallel for reduction(+: int_area)
+    KRATOS_ERROR_IF((std::abs( Factor ) - 1.0) > 1.0e-12) << "MassConservationPocess: Given value of argument 'factor' is not plausible. Consider '1' or '-1'." << std::endl;
+
+    #pragma omp parallel for reduction(+: outflow)
     for (int i_elem = 0; i_elem < static_cast<int>(mrModelPart.NumberOfElements()); ++i_elem){
         // iteration over all elements
         const auto it_elem = mrModelPart.ElementsBegin() + i_elem;
 
         Matrix shape_functions;
         GeometryType::ShapeFunctionsGradientsType shape_derivatives;
 
-        const auto rGeom = it_elem->GetGeometry();
+        auto r_geom = it_elem->GetGeometry();
         unsigned int pt_count_pos = 0;
         unsigned int pt_count_neg = 0;
-
-        // instead of using data.isCut()
-        for (unsigned int pt = 0; pt < rGeom.Points().size(); pt++){
-            if ( rGeom[pt].FastGetSolutionStepValue(DISTANCE) > 0.0 ){
-                pt_count_pos++;
-            } else {
-                pt_count_neg++;
-            }
-        }
-
-        if ( pt_count_pos == rGeom.PointsNumber() || pt_count_neg == rGeom.PointsNumber() ){
-            // all nodes are on one side and the element is not split
-            continue;
-        }
-        else {
+        const bool geom_is_cut = IsGeometryCut(r_geom, pt_count_neg, pt_count_pos);
+        if (geom_is_cut){
             // element is cut by the surface (splitting)
-            Kratos::unique_ptr<ModifiedShapeFunctions> p_modified_sh_func = nullptr;
             Vector w_gauss_interface(3, 0.0);
```

Comment:
```suggestion
            Vector w_gauss_interface;
```
This is not correct in 2D. Indeed, the modified shape functions will resize it.